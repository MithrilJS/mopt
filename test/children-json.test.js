"use strict";

var code = require("./lib/code");

describe("Children", () => {
    describe("JSON.prototype children", () => {
        it("should know that JSON.stringify is safe", () =>
            expect(
                code(`m("div", JSON.stringify({}))`)
            )
            .toMatchSnapshot()
        );
        
        it("shouldn't transform JSON.parse since it may not be safe", () =>
            expect(
                code(`m("div", JSON.parse({}))`)
            )
            .toMatchSnapshot()
        );
    });
});
