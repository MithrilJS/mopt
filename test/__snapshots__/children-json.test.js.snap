// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Children JSON.prototype children should know that JSON.stringify is safe 1`] = `"m.vnode(\\"div\\",undefined,undefined,undefined,JSON.stringify({}),undefined);"`;

exports[`Children JSON.prototype children shouldn't transform JSON.parse since it may not be safe 1`] = `"m(\\"div\\",JSON.parse({}));"`;
